### ISSUE TITLE
Support reading from streams/memory

### ISSUE THREAD

**Issue by silicakes** [photostructure/exiftool-vendored.js#84]
**Is your feature request related to a problem? Please describe.**
I'm trying to infer some exif metadata from s3 images without loading the whole thing.
I'm using about 512 bytes in order to find it.

Since I'm not writing the file anywhere or don't want to store it on disk, I prefer to pass a binary representation to get the data I need.

**Describe the solution you'd like**
Possibly accepting and traversing a binary format from memory:

```typescript

// I'm probably reaching with the `ReadableStream` 
// - it could be great for someone else, but I don't need it for my use case
const data: Buffer|Uint8Array|Blob|string|Readable = ...;

exiftool
  .readFromMemory(data)
  .then((tags /*: Tags */) =>
    console.log(
      `Make: ${tags.Make}, Model: ${tags.Model}, Errors: ${tags.errors}`
    )
  )
  .catch((err) => console.error("Something terrible happened: ", err))
```

**Describe alternatives you've considered**
Using the following from a rather old library: https://github.com/gomfunkel/node-exif/blob/master/lib/exif/ExifImage.js#L217

**Comment by mceachen** [comment]
TL:DR; ExifTool has a "-fast" and "-fast2" mode that you can enable in this library, which only reads the header metadata, and would avoid your need to send byte buffers to ExifTool. See the docs for `ExifTool.read` and https://exiftool.org/#performance.

Note that this library runs ExifTool in `stay_open` mode, which only works with filenames, rather than byte buffers. See https://exiftool.org/forum/index.php?topic=3518.msg15977#msg15977 

If still don't get satisfaction from `-fast2`, you can avoid normal I/O latency by reading and writing to a ramdisk like /dev/shm if your platform supports it.

**Comment by silicakes** [comment]
> TL:DR; ExifTool has a "-fast" and "-fast2" mode that you can enable in this library, which only reads the header metadata, and would avoid your need to send byte buffers to ExifTool. See the docs for `ExifTool.read` and https://exiftool.org/#performance.
> 
> Unless you're the producer of all images, the assumption that 512 bytes will be readable may not be safe. Tag encoding is infuriatingly disparate across makes, models, and libraries. (thank goodness for ExifTool!)
> 
> Note that this library runs ExifTool in `stay_open` mode, which only works with filenames, rather than byte buffers. See https://exiftool.org/forum/index.php?topic=3518.msg15977#msg15977
> 
> If still don't get satisfaction from `-fast2`, you can avoid normal I/O latency by reading and writing to a ramdisk like /dev/shm if your platform supports it.

Thank you for the comprehensive answer @mceachen .
I think I have everything I need to come up with my own solution for this!

**Comment by mceachen** [comment]
> Thank you for the comprehensive answer @mceachen .

:+1: 

>  I think I have everything I need to come up with my own solution for this!

@silicakes feel free to comment here or add to the wiki if you think your solution is applicable to others.
